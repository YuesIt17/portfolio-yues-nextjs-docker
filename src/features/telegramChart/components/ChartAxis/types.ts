import {TChartDataLabel} from '@/utils/types';

export type TChartAxis = {
  points: string;
  typeAxis: 'x' | 'y';
  labels: TChartDataLabel[];
  labelPaddingX?: number;
  labelPaddingY?: number;
  maxDataX?: number;
};
