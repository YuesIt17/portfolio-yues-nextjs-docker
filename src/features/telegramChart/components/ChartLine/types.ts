export type TChartLine = {
  points: string;
  fill?: string;
  stroke?: string;
  isVisible?: boolean;
};
