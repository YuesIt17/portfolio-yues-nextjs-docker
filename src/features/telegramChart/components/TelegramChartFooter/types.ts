import {TChartDataLine} from '@/utils/types';

export type TTelegramChartFooter = {
  data: TChartDataLine[] | undefined;
  onChange: (line: TChartDataLine, isVisible: boolean) => void;
};
