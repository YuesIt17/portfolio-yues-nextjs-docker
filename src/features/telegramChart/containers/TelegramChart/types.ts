export type TTelegramChart = {
  width: number;
};

export type TTelegramChartStyle = TTelegramChart;
