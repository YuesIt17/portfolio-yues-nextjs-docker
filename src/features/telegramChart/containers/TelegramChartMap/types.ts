import {TChartDataLine} from '@/utils/types';

export type TTelegramChartMap = {
  width: number;
  data: TChartDataLine[];
  maxDataX: number;
  maxDataY: number;
};

export type TTelegramChartMapStyle = {
  width: number;
};
