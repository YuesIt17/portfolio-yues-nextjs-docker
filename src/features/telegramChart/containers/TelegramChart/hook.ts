import {useEffect, useMemo} from 'react';

import {TChartDataLine} from '@/utils/types';
import {prepareData} from '@/utils';
import {useTypeDispatch, useTypeSelector} from '@/redux/hooks';
import {setAll, updateOne} from '../../store';
import {getDataChart} from '../../store/saga/actions';
import {selectChartData, selectChartLine} from '../../store/selectors';

export function useTelegramChart() {
  const dispatch = useTypeDispatch();
  const chartData = useTypeSelector(selectChartData);
  const chartLines = useTypeSelector(selectChartLine);

  const {lines, maxDataX, maxDataY, labelsX, labelsY} = useMemo(() => {
    return prepareData(chartData);
  }, [chartData]);

  useEffect(() => {
    dispatch(getDataChart());
  }, []);

  useEffect(() => {
    dispatch(setAll(lines));
  }, [lines]);

  const onChangeHadler = (line: TChartDataLine, isVisible: boolean) => {
    dispatch(updateOne({...line, isVisible: isVisible}));
  };

  return {
    data: chartLines,
    maxDataX,
    maxDataY,
    labelsX,
    labelsY,
    onChangeHadler,
  };
}
