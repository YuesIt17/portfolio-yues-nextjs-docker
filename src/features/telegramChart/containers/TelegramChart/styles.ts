import {css} from '@emotion/react';
import {TTelegramChartStyle} from './types';

export const useStyles = ({width}: TTelegramChartStyle) => ({
  chart: css({
    width,
    padding: '20px 0',
    alignSelf: 'flex-start',
  }),
});
