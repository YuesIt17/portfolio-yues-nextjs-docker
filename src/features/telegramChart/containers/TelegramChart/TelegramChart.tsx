import React from 'react';
import {useStyles} from './styles';
import {TTelegramChart} from './types';
import {useTelegramChart} from './hook';
import {TelegramChartFooter, TelegramChartLine} from '../../components';
import {TelegramChartMap} from '../TelegramChartMap/TelegramChartMap';
import {PEDDING_CHART_SIZE} from '../../constants';

export const TelegramChart = ({width}: TTelegramChart) => {
  const styles = useStyles({width});
  const {data, maxDataX, maxDataY, labelsX, labelsY, onChangeHadler} =
    useTelegramChart();

  if (!data) return null;
  return (
    <div css={styles.chart} data-testid="telegramChart">
      <TelegramChartLine
        data={data}
        maxDataX={maxDataX}
        maxDataY={maxDataY}
        labelsX={labelsX}
        labelsY={labelsY}
        peddingSize={PEDDING_CHART_SIZE}
      />
      <TelegramChartMap
        data={data}
        maxDataX={maxDataX}
        maxDataY={maxDataY}
        width={width}
      />
      <TelegramChartFooter data={data} onChange={onChangeHadler} />
    </div>
  );
};
