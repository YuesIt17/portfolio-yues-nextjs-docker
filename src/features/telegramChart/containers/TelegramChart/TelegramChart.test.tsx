import React from 'react';
import {themeTestRenderer} from '@/utils/tests';
import {TelegramChart} from './TelegramChart';
import {TChartDataLabel} from '@/utils/types';
import {mockChartDataLine} from '@/share/mockDataTest';

jest.mock('./hook.ts', () => ({
  useTelegramChart: jest.fn(() => ({
    data: mockChartDataLine,
    maxDataX: 11,
    maxDataY: 37,
    labelsX: [
      {
        label: 'Nov 17',
        coordinate: 1,
      },
    ] as TChartDataLabel[],
    labelsY: [
      {
        label: '60',
        coordinate: 23,
      },
    ] as TChartDataLabel[],
  })),
}));

describe('TelegramChart test', () => {
  test('Check render TelegramChart component', () => {
    const telegramChart = themeTestRenderer(
      <TelegramChart width={700} />
    ).toJSON();

    expect(telegramChart).toMatchSnapshot();
  });
});
