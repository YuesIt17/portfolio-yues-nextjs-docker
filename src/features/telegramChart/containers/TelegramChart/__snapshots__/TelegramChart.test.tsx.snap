// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegramChart test Check render TelegramChart component 1`] = `
<div
  className="css-1km59tu-chart"
  data-testid="telegramChart"
>
  <svg
    style={
      Object {
        "width": undefined,
      }
    }
    viewBox="0 0 41 67"
  >
    <g>
      <polyline
        className="css-1ied4zn-line"
        points="0,37 11,37"
      />
      <text
        className="css-1nvft8-label"
        x={1}
        y={61}
      >
        Nov 17
      </text>
    </g>
    <g>
      <polyline
        className="css-1ied4zn-line"
        points="0,0 0,37"
      />
      <g>
        <text
          className="css-1nvft8-label"
          x={0}
          y={11}
        >
          60
        </text>
        <polyline
          className="css-1ied4zn-line"
          points="0,23 11,23"
        />
      </g>
    </g>
  </svg>
  <div
    className="css-187vt5f-chartMap"
    data-testid="telegramChartMap"
  >
    <svg
      style={
        Object {
          "width": undefined,
        }
      }
      viewBox="0 0 11 7"
    />
  </div>
  <div
    className="css-1weht41-footer"
  >
    <button
      className="css-mrh4cr-button"
      data-testid="name_1"
      onClick={[Function]}
    >
      <div
        className="css-6o636l-icon"
      >
        <svg
          fill="transparent"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            fill="transparent"
            r="11"
            stroke="#F34C44"
            strokeWidth={2}
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </div>
      <div
        className="css-u4zqfx-label"
      >
        name_1
      </div>
    </button>
  </div>
</div>
`;
